packages/backend/src/ee/services/ai/utils/getChatHistoryFromThreadMessages.ts

```diff
@@ -0,0 +1,92 @@
+import {
+    AssistantContent,
+    CoreAssistantMessage,
+    CoreMessage,
+    CoreUserMessage,
+    TextPart,
+    ToolCallPart,
+} from 'ai';
+import { AiModel } from '../../../models/AiModel';
+
+export const getChatHistoryFromThreadMessages = (
+    // TODO: move getThreadMessages to AiAgentModel and improve types
+    // also, it should be called through a service method...
+    threadMessages: Awaited<
+        ReturnType<typeof AiModel.prototype.getThreadMessages>
+    >,
+) =>
+    threadMessages.flatMap<CoreMessage>((message) => {
+        const messages: CoreMessage[] = [
+            { role: 'user', content: message.prompt } satisfies CoreUserMessage,
```

Comment:
there is no need to use `satisfies` here (or when pushing to the array) right? as it will all be a `CoreMessage` element

Reply:
correct, was just helpful when I was coding :D